lib/mpool/src/mdc_file.c

```diff
@@ -302,15 +312,27 @@ mdc_file_validate(struct mdc_file *mfp, uint64_t *gen)
 
             err = logrec_validate(mfp, addr, &recsz);
             if (err) {
-                if (merr_errno(err) == ENOMSG) { /* End of log */
+                if (merr_errno(err) == ENODATA || merr_errno(err) == ENOMSG) { /* End of log */
                     err = 0;
                     mfp->woff = addr - mfp->addr;
                     break;
                 }
                 goto errout;
             }
 
-            addr += (rhlen + recsz);
+            switch (mfp->lh.vers) {
+            case MDC_LOGHDR_VERSION:
+                addr += (rhlen + ALIGN(recsz, alignof(uint64_t)));
```

Comment:
alignof seems a little weird, but i don't have time right now to investigate...

Reply:
Changed it to sizeof(uint64_t) to be clear and work correctly irrespective of the platform. Thnx!